api/envoy/config/route/v3/route_components.proto

```diff
@@ -890,6 +890,11 @@ message RouteAction {
     //   must come from trusted source.
     string host_rewrite_header = 29
         [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME strict: false}];
+
+    // Indicates that during forwarding, the host header will be swapped with
+    // the result of the regex substitution executed on path value with query and fragment removed.
+    // This is useful for transitioning variable content between path segment and subdomain.
```

Comment:
While it is clear from the test cases, I think giving an example here will be useful.